File: array/704_Binary_Search/704.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)  # [left ,right)

        while left < right:
            middle = left + (right - left) // 2
            if nums[middle] < target:
                left = middle + 1
            elif nums[middle] > target:
                right = middle
            else:
                return middle
        
        return -1   # Target value not found

File: array/704_Binary_Search/704.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length;
        while(left < right){
            int middle = left + ((right - left) >> 1);
            if (nums[middle] < target) {
                left = middle + 1;
            } else if (nums[middle] > target) {
                right = middle;
            } else {
                return middle;
            }
        }
        return -1;
    }
}